purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesOrchestrator.kt

```diff
@@ -214,6 +217,37 @@ internal class PurchasesOrchestrator constructor(
 
     // region Public Methods
 
+    fun syncAttributesAndOfferingsIfNeeded(
+        callback: SyncAttributesAndOfferingsCallback,
+    ) {
+        val receiveOfferingsCallback = object : ReceiveOfferingsCallback {
+            override fun onReceived(offerings: Offerings) {
+                callback.onSuccess(offerings)
+            }
+
+            override fun onError(error: PurchasesError) {
+                callback.onError(error)
+            }
+        }
+
+        if (!lastSyncAttributesAndOfferingsRateLimiter.shouldProceed()) {
+            log(
+                LogIntent.WARNING,
+                SyncAttributesAndOfferingsStrings.RATE_LIMIT_REACHED.format(
+                    lastSyncAttributesAndOfferingsRateLimiter.maxCalls,
+                    lastSyncAttributesAndOfferingsRateLimiter.periodSeconds,
+                ),
+            )
+
+            getOfferings(receiveOfferingsCallback)
+            return
+        }
+
+        subscriberAttributesManager.synchronizeSubscriberAttributesForAllUsers(appUserID) {
+            getOfferings(receiveOfferingsCallback, fetchCurrent = true)
```

Comment:
I'm wondering if we should fallback to the cached offerings if the `getOfferings` call fails... Considering we fallback to the cached offerings before, I think it would be better... and maybe add a log.

Reply:
I _think_ this already happens in `fetchAndCacheOfferings()` 🤔 If there is a server error, `offeringsCache.cachedOfferingsResponse` is used. Its not the "in memory" cache but it should be the same thing. Thoughts?

Reply:
Oh right 🤦 . Please ignore me :P